artiq/test/coredevice/test_rtio.py

```diff
@@ -189,6 +189,53 @@ def run(self):
         self.set_dataset("count", self.loop_in.count())
 
 
+class LoopbackGateTiming(EnvExperiment):
+    def build(self):
+        self.setattr_device("core")
+        self.setattr_device("loop_in")
+        self.setattr_device("loop_out")
+
+    @kernel
+    def run(self):
+        # Make sure there are no leftover events.
+        self.core.break_realtime()
+        self.loop_in.count()
```

Comment:
There may be leftover events because of the very bug you are testing against (due to the gateware latency in the timestamp counter to gate path, the timeout - referenced to the timestamp counter - is too small and the kernel thinks the gateware has closed the gate when it has not). Use ``core.reset()`` instead.